src/Components/Web/src/Forms/InputFile.cs

```diff
@@ -35,6 +35,18 @@ public class InputFile : ComponentBase, IInputFileJsCallbacks, IDisposable
         [Parameter]
         public EventCallback<InputFileChangeEventArgs> OnChange { get; set; }
 
+        /// <summary>
+        /// Gets or sets the maximum allowed file size in bytes. Defaults to 15 MB.
+        /// </summary>
+        [Parameter]
+        public long MaxFileSize { get; set; } = 1024 * 1024 * 15;
+
+        /// <summary>
+        /// Gets or sets the maximum allowed number of files. Defaults to 3.
+        /// </summary>
+        [Parameter]
+        public int MaxAllowedFiles { get; set; } = 3;
```

Comment:
3 is an awkward default because a developer might realistically test uploading 2 or 3 files and believe that they support an arbitrary number. Personally I think that either:

 * We should change this to a much higher value just to catch extreme edge cases, rather than interfering with typical use cases. For example, to 50 or 100.
 * Or we should change it to 1, so developers who intend to support multiple file uploads will certainly hit it in their most basic local verification, and then know they have to pick a value themselves.